workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,460 @@
+
+  # Each env has their own stakeholder approval environment.  If no required reviewers are set for
+  # that environment, the workflow will continue without requiring anyone to approve the deployment.
+  stakeholder-approval:
+    needs: [set-vars]
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+    environment: '${{ needs.set-vars.outputs.ENVIRONMENT }} Stakeholder Approval'
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      - name: Approval Received
+        run: echo "Stakeholder approval was received"
+
+  # This job needs to run for all environments because tf-plan relies
+  # on it but the steps inside this job will only run for the Prod env.
+  validate-tag-for-prod-deploys:
+    needs: [set-vars, stakeholder-approval]
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+
+      # In this job, always checkout the default branch (not the branch that was provided as an input).  Also use
+      # fetch-depth: 0 to retrieve the history and tags so we can check if a tag is reachable from the default branch.
+      - name: Checkout Repository
+        if: needs.set-vars.outputs.ENVIRONMENT == 'prod'
+        uses: actions/checkout@v2
+        with:
+          ref: 'main' # TODO: verify the name of your default branch
+          fetch-depth: 0
+
+      - uses: im-open/is-tag-reachable-from-default-branch@v1.0.0
+        if: needs.set-vars.outputs.ENVIRONMENT == 'prod'
+        with:
+          tag: ${{ env.GITHUB_REF }}
+
+  build-deploy:
+    needs: [tech-approval, qa-approval, stakeholder-approval] # TODO: Update for the approval environments used
+    runs-on: [self-hosted, windows-2019]
+    environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+
+    env:
+      PAGERDUTY_SERVICE_IDS: '[ "${{secrets.PAGERDUTY_SERVICE_ID}}" ]' # TODO:  Add your PD Service IDs here, the svc id usually has a format of P0XXXXX
+      PAGERDUTY_WINDOW_IN_MIN: 30 # TODO:  Verify the length of your PD Maintenance Window
+      PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ needs.set-vars.outputs.ENVIRONMENT }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+      DEPLOY_BOARD_NUM: '' # TODO: Add the automated deployment board number or remove if not using an automated deployment project board.
+      DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+      PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+      ARTIFACT_NAME: 'Published Site' # TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+      ARTIFACT_ZIP: 'artifacts.zip'
+      DEPLOY_ZIP: 'published_site.zip'
```

The DEPLOY_BOARD_NUM var isn't used in this job and it was replaced with a TODO in the deploy board step.
The artifact name/zip shouldn't be used.  We took that step out of the corresponding app service/function deploy workflows.

```suggestion
      DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
      PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
      DEPLOY_ZIP: 'published_site.zip'
```